ts/sagas/startup/watchApplicationActivitySaga.ts

```diff
@@ -42,41 +44,63 @@ export function* watchApplicationActivitySaga(): IterableIterator<Effect> {
-    ) {
-      // Cancel the background timer if running
-
-      yield cancel(identificationBackgroundTimer);
-      identificationBackgroundTimer = undefined;
-
-      // Check if there is a payment ongoing
-      const isPaymentOngoing: ReturnType<
-        typeof isPaymentOngoingSelector
-      > = yield select<GlobalState>(isPaymentOngoingSelector);
-
-      // Check if we have a pending notification message
-      const pendingMessageState: ReturnType<
-        typeof pendingMessageStateSelector
-      > = yield select<GlobalState>(pendingMessageStateSelector);
-
-      // We only navigate to the new message from a push if we're not in a
-      // payment flow
-      if (!isPaymentOngoing && pendingMessageState) {
-        // We have a pending notification message to handle
-        const messageId = pendingMessageState.id;
-
-        // Remove the pending message from the notification state
-        yield put(clearNotificationPendingMessage());
-
-        // Navigate to message details screen
-        yield put(navigateToMessageDetailScreenAction({ messageId }));
+    } else if (lastState !== "active" && newApplicationState === "active") {
+      if (identificationBackgroundTimer) {
+        // Cancel the background timer if running
+        yield cancel(identificationBackgroundTimer);
+        identificationBackgroundTimer = undefined;
       }
+
+      yield call(watchRequiredIdentificationSaga);
+
+      yield call(watchNotificationSaga);
     }
 
     // Update the last state
     lastState = newApplicationState;
   });
 }
+
+/**
+ * Check if the user presses on a notification and redirect to the
+ * related message if any payment flow has been started
+ */
+function* watchNotificationSaga(): SagaIterator {
+  // Check if there is a payment ongoing
+  const isPaymentOngoing: ReturnType<
+    typeof isPaymentOngoingSelector
+  > = yield select<GlobalState>(isPaymentOngoingSelector);
+
+  // Check if we have a pending notification message
+  const pendingMessageState: ReturnType<
+    typeof pendingMessageStateSelector
+  > = yield select<GlobalState>(pendingMessageStateSelector);
+
+  // We only navigate to the new message from a push if we're not in a
+  // payment flow
+  if (!isPaymentOngoing && pendingMessageState) {
+    // We have a pending notification message to handle
+    const messageId = pendingMessageState.id;
+
+    // Remove the pending message from the notification state
+    yield put(clearNotificationPendingMessage());
+
+    // Navigate to message details screen
+    yield put(navigateToMessageDetailScreenAction({ messageId }));
+  }
+}
+
+function* watchRequiredIdentificationSaga() {
+  // wallet screens requiring identification before they can be displayed when app goes background
+  const whiteList: ReadonlyArray<any> = [ROUTES.WALLET_ADD_CARD];
+
+  const nav: ReturnType<typeof navSelector> = yield select<GlobalState>(
+    navSelector
+  );
+  const walletNav = nav.routes[0].routes[1].routes;
```

Comment:
```suggestion
// wallet screens requiring identification before they can be displayed when app goes in background
  const nav: ReturnType<typeof navSelector> = yield select<GlobalState>(
    navSelector
  );
  if (getCurrentRouteName(nav) === ROUTES.WALLET_ADD_CARD) {
    // Request identification to display the screen
    yield put(identificationRequest());
  }
```
we already have an util function **_getCurrentRouteName_** for this purpose

Reply:
https://github.com/pagopa/io-app/pull/1537/commits/2d7f572e5041e59f1074582865a0a02417ce4bdc isolate the existing function as a navigation util.

https://github.com/pagopa/io-app/pull/1537/commits/775068416fad0fa4938ea9520a887868cf66b785 uses the existing function for the implementation.  Moreover, the same commit introduces a refactoring to isolate:
- the code related to the redirect from the notification to the message detail (`watchNotificationSaga`)
- the code to manage the request of identification when the app state changes (`watchRequiredIdentificationSaga`)